### PR TITLE
Issue #2664676: Add test cases for empty store

### PR DESCRIPTION
**Pull request by miteshmap**
Added new test classes. OrderEmptyStoreTest, ProductEmptyStoreTest.
based on comments of - https://github.com/drupalcommerce/commerce/pull/345

### CODE REVIEW

File: modules/order/src/Tests/OrderEmptyStoreTest.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+ */
+class OrderEmptyStoreTest extends CommerceTestBase {
+
+  /**
+   * Modules to enable.
+   *
+   * @var array
+   */
+  public static $modules = [
+    'commerce_product',
```
Comment (mglaman) [inline modules/order/src/Tests/OrderEmptyStoreTest.php]: Do we need commerce_product? Nitpick to try and keep tests a _little_ faster by enabling less modules.

File: modules/product/src/Tests/ProductEmptyStoreTest.php [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+   *
+   * @var array
+   */
+  public static $modules = [
+    'commerce_store',
+    'commerce_product',
+    'commerce_order',
+    'field_ui',
+    'options',
+    'taxonomy',
```
Comment (mglaman) [inline modules/product/src/Tests/ProductEmptyStoreTest.php]: Do we need commerce_order, field_ui, options, or taxonomy? Nitpick to try and keep tests a _little_ faster by enabling less modules.

**Comment by mglaman** [comment]
@bojanz I think this looks good now

**Comment by mglaman** [comment]
We need to rebase this and make it a Functional test.

**Comment by bojanz** [comment]
Continued in #461.
